src/build/__init__.py

```diff
@@ -139,12 +141,29 @@ class ProjectBuilder(object):
     The PEP 517 consumer API.
     """
 
-    def __init__(self, srcdir, python_executable=sys.executable, scripts_dir=None):
-        # type: (str, Union[bytes, Text], Optional[Union[bytes, Text]]) -> None
+    def __init__(
+        self,
+        srcdir,  # type: str
+        python_executable=sys.executable,  # type: Union[bytes, Text]
+        scripts_dir=None,  # type: Optional[Union[bytes, Text]]
+        runner=pep517.default_subprocess_runner,  # type: RUNNER_TYPE
+    ):
+        # type: (...) -> None
         """
         :param srcdir: The source directory
         :param scripts_dir: The location of the scripts dir (defaults to the folder where the python executable lives)
         :param python_executable: The python executable where the backend lives
+        :param runner: An alternative runner for backend subprocesses
+
+        The 'runner', if provided, must expect the following:
+            - cmd: a list of strings representing the command and arguments to
+              execute, as would be passed to e.g. 'subprocess.check_call'.
+            - cwd: a string representing the working directory that must be
+              used for the subprocess. Corresponds to the provided source_dir.
```

Comment:
Done. Sorry, I just copy/pasted the docs from `pep517.wrappers`. I should have taken more care to check they still made sense. I hope the unindent worked, I can't easily preview the docs build unfortunately - the files in `.tox\docs\tmp` seem to refer to files in a `_static` directory which isn't built by `tox -e docs`.

Reply:
Hmm, I wonder if the lack of `_static` is because `tox -e docs` fails for me with an error:

```
C:\Work\Projects\build\docs\index.rst:33:toctree contains reference to document 'changelog' that doesn't have a title: no link will be generated
```

That's not something I changed, so I guess it's either a known issue, or it's something Windows-specific (which seems unlikely)?

Reply:
I assume it had something to do with the symlink?  The docs are also generated for every PR on Read The Docs:  https://pypa-build--290.org.readthedocs.build/en/290/

Reply:
Oh, if it's a symlink, then yes that would have messed up badly, as git doesn't seem to support symlinks on Windows (even though Windows does). I hope I haven't accidentally converted it to a normal file...

Reply:
Still a symlink by the look of it - at least it appears that way on GH and it is for me locally.